Swik 1524 refactor components made by kurt

TBoonX opened the pull request:
I removed the direct usage of the UserProfileSTore in components I created if it was possible.

rmeissn commented on components/Login/LoginModal.js:
```diff
@@ -65,18 +64,30 @@ class LoginModal extends React.Component {
+        localStorage.setItem(MODI, 'login_failed_register_now');
+
+        let thatContext = this.context;
+        async.series([
```
Why use an async.series to execute one action?

rmeissn commented on components/Login/LoginModal.js:
```diff
@@ -65,18 +64,30 @@ class LoginModal extends React.Component {
+
+    componentDidUpdate() {
+        // console.log('componentDidUpdate:', this.props.errorMessage, this.props.socialLoginError, this.props.userid, this.props.username);
+        if (this.props.errorMessage !== '' && this.props.errorMessage !== undefined && this.isLoading) {
```
You could use `!common.isEmpty()` instead of these two statements `this.props.errorMessage !== '' && this.props.errorMessage !== undefined`

rmeissn commented on components/Login/LoginModal.js:
```diff
@@ -65,18 +64,30 @@ class LoginModal extends React.Component {
-            $('.ui.login.modal').modal('hide');
-        }
-        if (localStorage.getItem(MODI) === 'login'&& nextProps.UserProfileStore.socialLoginError){
+        else if (localStorage.getItem(MODI) === 'login' && this.props.socialLoginError){
```
Don't you need to test the prop for emptiness?

TBoonX commented:
@rmeissn Please review again and test LoginModal and everything under UserProfile

kadevgraaf commented:
Are the requested changes/review ok?

rmeissn commented:
I had no time to review the latest changes yet. I will do so this week. It's needed to click through ALL of the user-profile stuff in slidewiki-platform as this code touches most of the functionalities.

rmeissn commented:
I tested the feature by trying to test every single user profile related action:
* When I log in, before the site reloads, an error message is printed to the modal, just containing a dot
* When I try to enable my Github Account for my profile, the window won't close and displays `error=Grant%3A%20missing%20session%20or%20misconfigured%20provider`
* It's the same for G+
* I can't enable Facebook, as it quits with another error message
* When I try to login for an deactivated user, an error message is printed, containing just a dot
* If I try again with another user, it responds with 404, but the spinners on username and password won't stop

TBoonX commented:
@rmeissn 
* fixed
* Correct because social sign in on localhost is not possible - I am not able to close the window when such an error happens (also the loading indicator ist still in the settings view)
* Correct because social sign in on localhost is not possible
* Where did you find facebook? I removed it months ago
* Fixed
* Fixed

rmeissn commented:
Thanks for the fast fixes! I will do a new review next week.

I've found Facebook in my user-profile settings as a social provider.

rmeissn commented:
I've tested the whole user-stuff again. Seems to work fine. 
**Two minor issues I spotted**
For a newly created user: if I go to "my notifications", a "loading" message is displayed all the time. See

![loading](https://user-images.githubusercontent.com/855967/31272135-33387722-aa8a-11e7-91cb-05d5e5faf4a2.png)

If I open up the header bar user dropdown, it's displayed horizontally, instead of vertically on a 23" 16:9 Full-HD display. See: 

![user bar](https://user-images.githubusercontent.com/855967/31272189-6161610e-aa8a-11e7-9f67-1c0bcc3a5172.png)
